internal/consensus/state.go

```diff
@@ -2515,12 +2515,12 @@ func (cs *State) calculatePrevoteMessageDelayMetrics() {
 		_, val := cs.Validators.GetByAddress(v.ValidatorAddress)
 		votingPowerSeen += val.VotingPower
 		if votingPowerSeen >= cs.Validators.TotalVotingPower()*2/3+1 {
-			cs.metrics.QuorumPrevoteMessageDelay.Set(v.Timestamp.Sub(cs.Proposal.Timestamp).Seconds())
+			cs.metrics.QuorumPrevoteMessageDelay.With("proposer_address", cs.Validators.GetProposer().Address.String()).Set(v.Timestamp.Sub(cs.Proposal.Timestamp).Seconds())
```

Comment:
Is this really a network address, or is it something more durable like a public key? We probably do _not_ want to put a network address in a metric label. Besides being of high cardinality (higher than the size of the network, over time), it would potentially produce weird (spurious) collisions in stored time series.

Not that public keys can't vary too, but I expect they do so at a much slower rate, and there's some cost to swapping them.

Reply:
Great question. This is a public key. I definitely agree that network address would be much too high cardinality to put here. This should roughly be bounded by the size of the validator set, although it will increase as new validators are added. 

It's the string method of this type:
https://github.com/tendermint/tendermint/blob/4aa2184baee3cd000e2b5618f0bbc2c2512c1b40/crypto/crypto.go#L17